UnitsNet/CustomCode/Wrappers/Pressure.Wrapper.cs

```diff
@@ -0,0 +1,194 @@
+        private double AsBaseNumericType(PressureReference reference)
+        {
+            var baseReferenceValue = AsBaseReference();
+
+            if (Reference == reference)
+            {
+                return _pressure.Value;
+            }
+
+            var negatingValue = Reference == PressureReference.Vacuum ? -1 : 1;
+
+            switch (reference)
+            {
+                case PressureReference.Absolute: return baseReferenceValue;
+                case PressureReference.Gauge: return baseReferenceValue - AtmosphericPressure.ToUnit(_pressure.Unit).Value;
+                case PressureReference.Vacuum: return AtmosphericPressure.ToUnit(_pressure.Unit).Value - negatingValue * baseReferenceValue;
+                default:
+                    throw new NotImplementedException($"Can not convert {Reference} to {reference}.");
+            }
+        }
+
+        /// <summary>
+        ///     Converts <see cref="Pressure.Value" /> at <see cref="Reference" /> to <see cref="double" /> at
+        ///     <see cref="BaseReference" />
+        /// </summary>
+        /// <returns>The value of pressure at the <see cref="BaseReference" /></returns>
+        private double AsBaseReference()
+        {
+            switch (Reference)
+            {
+                case PressureReference.Absolute:
+                {
+                    if (_pressure.Value < 0)
+                    {
+                        throw new ArgumentOutOfRangeException("Absolute pressure cannot be less than zero.");
+                    }
+
+                    return _pressure.Value;
+                }
+                case PressureReference.Gauge:
+                {
+                    if (_pressure.Value * -1 > AtmosphericPressure.ToUnit(_pressure.Unit).Value)
+                    {
+                        throw new ArgumentOutOfRangeException("Absolute pressure cannot be less than zero.");
+                    }
+
+                    return AtmosphericPressure.ToUnit(_pressure.Unit).Value + _pressure.Value;
+                }
+                case PressureReference.Vacuum:
+                {
+                    if (_pressure.Value > AtmosphericPressure.ToUnit(_pressure.Unit).Value)
+                    {
+                        throw new ArgumentOutOfRangeException("Absolute pressure cannot be less than zero.");
+                    }
+
+                    return AtmosphericPressure.ToUnit(_pressure.Unit).Value - _pressure.Value;
+                }
+                default:
+                    throw new NotImplementedException($"Can not convert {Reference} to base reference.");
+            }
+        }
+
+        /// <summary>
+        ///     Represents the pressure at which _pressure is referenced (1 atm default)
+        /// </summary>
+        public static Pressure AtmosphericPressure { get; set; }
+
+        private static readonly Pressure _defaultAtmosphericPressure = new Pressure(1, PressureUnit.Atmosphere);
```

Comment:
Nitpick, but please move static fields to the top of the class.

We try to follow the common C# conventions pretty much as described here, but we're not super strict about it.
https://stackoverflow.com/a/310967/134761

Also, static fields should use `PascalCase` even if private.